recipes/libcap/all/conanfile.py

```diff
@@ -26,82 +30,81 @@ class LibcapConan(ConanFile):
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        tc.fpic = self.options.get_safe("fPIC", True)
+        env = tc.environment()
+        env.define("SHARED", "yes" if self.options.shared else "no")
+        env.define("PTHREADS", "yes" if self.options.psx_syscals else "no")
+        env.define("DESTDIR", self.package_folder)
+        env.define("prefix", "/")
+        env.define("lib", "lib")
+
+        if cross_building(self) and not env.vars(self).get("BUILD_CC"):
+            native_cc = "cc"
             self.output.info("Using native compiler '{}'".format(native_cc))
-            self._autotools_env["BUILD_CC"] = native_cc
+            env.define("BUILD_CC", native_cc)
 
-        return self._autotools, self._autotools_env
-
-    def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        tc.generate(env)
 
     def build(self):
-        self._patch_sources()
+        apply_conandata_patches(self)
 
-        with tools.chdir(os.path.join(self._source_subfolder, self.name)):
-            env_build, env_build_vars = self._configure_autotools()
-            env_build.make(vars=env_build_vars)
+        autotools = Autotools(self)
+        with chdir(self, os.path.join(self.source_folder, self.name)):
+            autotools.make()
 
     def package(self):
-        self.copy("License", dst="licenses", src=self._source_subfolder)
-
-        with tools.chdir(os.path.join(self._source_subfolder, self.name)):
-            env_build, env_build_vars = self._configure_autotools()
+        copy(self, "License", self.source_folder,
+             os.path.join(self.package_folder, "licenses"), keep_path=False)
 
-            env_build.make(target="install-common-cap", vars=env_build_vars)
+        autotools = Autotools(self)
+        with chdir(self, os.path.join(self.source_folder, self.name)):
+            autotools.make(target="install-common-cap")
             install_cap = ("install-shared-cap" if self.options.shared
                            else "install-static-cap")
-            env_build.make(target=install_cap, vars=env_build_vars)
+            autotools.make(target=install_cap)
 
             if self.options.psx_syscals:
-                env_build.make(target="install-common-psx",
-                               vars=env_build_vars)
+                autotools.make(target="install-common-psx")
                 install_psx = ("install-shared-psx" if self.options.shared
                                else "install-static-psx")
-                env_build.make(target=install_psx, vars=env_build_vars)
+                autotools.make(target=install_psx)
 
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
 
     def package_info(self):
+        self.cpp_info.components["cap"].set_property("pkg_config_name",
```

Comment:
When I'm trying to build a test package with this line I get a warning: "(test package): WARN: [libcap/2.65] The PC package name libcap.pc already exists and it matches with another component one. Please, review all the component's pkg_config_name defined. Skipping it!". But I don't know how to figure out what's wrong here.

Reply:
It's because there's a global one by default for v1 generators.

You need to set a root level one like AFAIK (not an expert just say that in another PR I reviewed recently)

```py
self.cpp_info.set_property("pkg_config_name", "libcap-all-do-not-use")
```